packages/open-payments/src/client/requests.ts

```diff
@@ -158,90 +199,113 @@ const checkUrlProtocol = (deps: BaseDeps, url: string): string => {
   return requestUrl.href
 }
 
-export const createAxiosInstance = (args: {
+type CreateHttpClientArgs = {
   requestTimeoutMs: number
-  privateKey?: KeyObject
-  keyId?: string
-}): AxiosInstance => {
-  const axiosInstance = axios.create({
+} & (
+  | { privateKey?: KeyObject; keyId?: string }
+  | { authenticatedRequestInterceptor: InterceptorFn }
+)
+
+export type HttpClient = KyInstance
+
+export type InterceptorFn = (request: Request) => Request | Promise<Request>
+
+export const createHttpClient = async (
+  args: CreateHttpClientArgs
+): Promise<HttpClient> => {
+  const { default: ky } = await import('ky')
+
+  const kyInstance = ky.create({
+    timeout: args?.requestTimeoutMs,
```

Comment:
```suggestion
    timeout: args.requestTimeoutMs,
```

no optional chaining necessary as far as I can tell

Reply:
Yes, thanks. Updating